lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -55,6 +57,17 @@ def is_proctored_exam_for_user(block_key):
                     unicode(block_key.course_key),
                     unicode(block_key),
                 )
-                return user_exam_summary and user_exam_summary['status'] != ProctoredExamStudentAttemptStatus.declined
+                exam_is_proctored = user_exam_summary and \
+                    user_exam_summary['status'] != ProctoredExamStudentAttemptStatus.declined
 
-        return [block_structure.create_removal_filter(is_proctored_exam_for_user)]
+                # after checking for proctored, we check for all other milestones
+                # if we need additional fields bump up version number
+                milestones_exist_for_exam = bool(milestones_helpers.get_course_content_milestones(
+                    unicode(block_key.course_key),
+                    unicode(block_key),
+                    'requires',
+                    usage_info.user.id
+                ))
+                return exam_is_proctored or milestones_exist_for_exam
```

Comment:
Nit: if you create separate internal helpers for `_is_exam_proctored` and `do_milestones_exist`, then it also optimizes the code - since the second call won't be made if the first succeeds.